### ISSUE TITLE
[qt5-base[core]:x64-osx] build failure

### ISSUE THREAD

**Issue by QuintusOostendorp** [microsoft/vcpkg#10209]
**Host Environment**
- OS: macOS 10.14.4
- Compiler: Apple clang version 11.0.0 (clang-1100.0.33.17)

**To Reproduce**
`./vcpkg install qt5`

**Failure logs** 

**Console output:**
```
Building package qt5-base[core]:x64-osx...
-- Figuring out qt target mkspec. Target arch x64
-- Target mkspec set to: macx-clang
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using cached /Users/thor/Development/vcpkg/downloads/qtbase-everywhere-src-5.12.5.tar.xz
-- Using source at /Users/thor/Development/vcpkg/buildtrees/qt5-base/src/5.12.5-009d7e4128
-- Detected OSX SDK Version: 10.15
-- Major.Minor OSX SDK Version: 10.15
-- Enviromnent OSX SDK Version: 10.15
-- Configuring x64-osx-dbg
-- Configuring x64-osx-dbg done
-- Configuring x64-osx-rel
-- Configuring x64-osx-rel done
-- NUMBER_OF_PROCESSORS is 8
-- Cleaning before build x64-osx-dbg
-- Building x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: /usr/bin/make -j8
    Working Directory: /Users/thor/Development/vcpkg/buildtrees/qt5-base/x64-osx-dbg
    Error code: 2
    See logs for more information:
      /Users/thor/Development/vcpkg/buildtrees/qt5-base/build-x64-osx-dbg-out.log
      /Users/thor/Development/vcpkg/buildtrees/qt5-base/build-x64-osx-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/install_qt.cmake:91 (vcpkg_execute_required_process)
  ports/qt5-base/portfile.cmake:265 (install_qt)
  scripts/ports.cmake:90 (include)


Error: Building package qt5-base:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: qt5-base:x64-osx
  Vcpkg version: 2020.02.04-unknownhash-external
```
**Contents of the build error log:**
```
ld: warning: directory not found for option '-L/Users/thor/Development/vcpkg/installed/x64-osx/debug/lib/manual-link'
ld: warning: directory not found for option '-L/Users/thor/Development/vcpkg/installed/x64-osx/debug/lib/manual-link'
ld: warning: direct access in function 'QHash<QString, int>::insert(QString const&, int const&)' from file '/Users/thor/Development/vcpkg/buildtrees/qt5-base/x64-osx-dbg/lib/li$
ld: warning: direct access in function 'QHash<QString, int>::insert(QString const&, int const&)' from file '/Users/thor/Development/vcpkg/buildtrees/qt5-base/x64-osx-dbg/lib/li$
ld: warning: direct access in function 'QHash<QString, int>::insert(QString const&, int const&)' from file '/Users/thor/Development/vcpkg/buildtrees/qt5-base/x64-osx-dbg/lib/li$
ld: warning: direct access in function 'QCommandLineParserPrivate::~QCommandLineParserPrivate()' from file '/Users/thor/Development/vcpkg/buildtrees/qt5-base/x64-osx-dbg/lib/li$
ld: warning: directory not found for option '-L/Users/thor/Development/vcpkg/installed/x64-osx/debug/lib/manual-link'
Sorry, "rcc" can not be run on this version of macOS. Qt requires macOS 10.15.0 or later, you have macOS 10.14.4.
make[2]: *** [.rcc/qrc_mimetypes.cpp] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [sub-corelib-make_first] Error 2
make: *** [sub-src-make_first] Error 2
```
**Additional context**
I did update vcpkg to the latest version and I've also installed the latest version of Xcode and the CLI tools. 

When I installed vcpkg I ran into this issue: https://github.com/microsoft/vcpkg/issues/9029
I fixed it by installing it with the command "CXXFLAGS="-D_CTERMID_H_" ./bootstrap-vcpkg.sh". Don't know if this might be related.

**Comment by JackBoosY** [comment]
@Neumann-A Could you take a look?

Thanks.

**Comment by Neumann-A** [comment]
@JackBoosY : Since I don't have an OSx system I can not say a lot about this error. But according to the error message:
`Sorry, "rcc" can not be run on this version of macOS. Qt requires macOS 10.15.0 or later, you have macOS 10.14.4.`

I qmake/configure somehow detects the wrong SDK version while building.
On the other hand the logic I implemented within the portfile detected 10.15:
`-- Detected OSX SDK Version: 10.15`

So the trick is here to figure out why there is a version mismatch between the logic in the portfile and the internal build of qt. Probably a search for "`can not be run`" in the source and builddir can shed more light on the error. And also having the configure logs could help.

**Comment by ThorThorsen** [comment]
Thanks for you comment, it gave me enough clues to figure it out.

In "portfile.cmake" the command "xcrun --show-sdk-version" is executed to determine the macOS sdk version. The QMAKE_MACOSX_DEPLOYMENT_TARGET environment variable is set with this value.

However, this command only returns the macOS sdk version number, which is not necessarily the version of the currently installed OS. In my case it returns 10.15 while my OS is on 10.14. 

I fixed it by adding "set(VCPKG_OSX_DEPLOYMENT_TARGET "10.14")" to the x64-osx triplet file.

Perhaps the xcrun command should be replaced by something like "sw_vers -productVersion | cut -d '.' -f 1,2"? This does return the currently installed macOS version.

**Comment by lz1998** [comment]
How to solve this problem?

**Comment by JackBoosY** [comment]
Use command `sw_vers -productVersion` can show the macOS system version.

**Comment by JackBoosY** [comment]
Hi guys, I'm trying to fix this issue in https://github.com/microsoft/vcpkg/pull/21486, can anyone test that?

Thanks.

**Comment by JackBoosY** [comment]
Fixed by #21486.
